src/overtone/sc/sclang.clj

```diff
@@ -0,0 +1,410 @@
+(ns overtone.sc.sclang
+  (:require
+   [babashka.process :as proc]
+   [clojure.edn :as edn]
+   [clojure.java.io :as io]
+   [clojure.string :as str]
+   [overtone.helpers.system :refer [get-os linux-os? mac-os? windows-os?]]
+   [overtone.sc.machinery.server.connection :as ov.conn]
+   [overtone.sc.synth :as ov.synth]))
+
+(defn sclang-path
+  "The user must have SuperCollider installed.
+
+  For sclang location in each OS, see
+  https://github.com/supercollider/supercollider/wiki/Path-searching#scide-finds-sclang"
+  []
+  (cond
+    (mac-os?)
+    "/Applications/SuperCollider.app/Contents/MacOS/sclang"
+    ;; It's simply `sclang` in non OSX environments,
+    ;; https://github.com/supercollider/supercollider/wiki/Path-searching#scide-finds-sclang
+    ;; TODO Someone should test on Linux and Windows to confirm that this works.
+    (windows-os?)
+    "sclang.exe"
```

Comment:
FWIW might be some ideas to copy from the functions that look for `scsynth`: https://github.com/overtone/overtone/blob/f5d4a82828b026bd470b9c2312ff488f565bbf98/src/overtone/sc/machinery/server/connection.clj#L276

Perhaps a last-ditch effort is to look at the `:sc-path` in the user config and search the same directory.

Reply:
Yeah, I've seen that, but I don't think we can be as flexible with `sclang` as we can with `scsynth`, `sclang` requires much more stuff to be in a working state and I am betting that's better to expect the user to have SC installed.

I find that's better to be guided by usercases that people complain about. I also didn't test it on Linux or Windows, if you have one of these and have some time (it should be 5 to 10 mins) to evaluate the code, start overtone and check that the last comment is working, it would help me \o

I've also added `*sclang-user-path` for arbritary paths and then we can enhance over time.